app/controllers/news_controller.rb

```diff
@@ -17,7 +17,9 @@ def index
   def resource_collection
     documents = Mas::Cms::News.all(
       params: { document_type: [NEWS_DOCUMENT_TYPE] }.merge(year_param)
-    )
+    ).map do |document|
+      TaggedNews.new(document)
+    end.sort_by(&:published_date).reverse
```

Comment:
There is a helluva lot going on in this method and personally I'm not a huge fan of chaining methods in this way. Perhaps a different method to manipulate the documents into the model and order you require would render this more readable?

Reply:
:recycle: